### PR TITLE
Fix event list reloading when coming back from event details. #273

### PR DESCRIPTION
**Pull request by lemonboston**
---

This will need to be rebased on `categories/master` first after #72 is closed.

This fix is only applicable now to retained fragments, if we encounter any other problem with normal fragments popped from back stack, we need to revisit the general topic. @dmfs the spreadsheet I shared with you contains I think a good overview of the lifecycles, so we can look into that again, if needed.

I've also changed the UP navigation in the Details MF to fire the BackTransition as you said.

### CODE REVIEW

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/fragments/EventDetailFragment.java
Comment (dmfs) [inline eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/fragments/EventDetailFragment.java]: Use `MicroFragmentEnvironment` for the variable type.

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/common/BaseBackStackedRetainedFragment.java
Comment (dmfs) [inline eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/common/BaseBackStackedRetainedFragment.java]: I'm not a big fan of the template method pattern.
Also, I consider this only a temporary workaround due to the lack of a better backstack implementation. No other Fragment we create should need this, so think it's not worth creating a separate abstraction class for this. That only encourages using this more often when we really should search for a better solution instead.

**Review by dmfs — CHANGES_REQUESTED** [review]

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/fragments/EventDetailFragment.java
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/fragments/EventDetailFragment.java]: Done.

File: eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/common/BaseBackStackedRetainedFragment.java
Comment (lemonboston) [reply eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/common/BaseBackStackedRetainedFragment.java]: Ok, I removed the base class and use the logic in `EventListListFragment` instead.

**Comment by lemonboston** [comment]
Changes done, but I suggest we wait until #272 is merged, then I squash and rebase on `categories/master` so you can have a final look.

**Comment by dmfs** [comment]
please resolve conflicts and rebase

**Comment by lemonboston** [comment]
Resolved and rebased.

**Review by dmfs — APPROVED** [review]
Looks good and appears to work well.

**Pull request merged**
